internal/pkg/db/mongo/metadata.go

```diff
@@ -557,12 +557,12 @@ func (mc MongoClient) UpdateDeviceProfile(dp contract.DeviceProfile) error {
 
 // Get the device profiles that are currently using the command
 func (mc MongoClient) GetDeviceProfilesUsingCommand(c contract.Command) ([]contract.DeviceProfile, error) {
-	query, err := idToBsonM(c.Id)
+	command, err := mc.getCommandById(c.Id)
 	if err != nil {
 		return []contract.DeviceProfile{}, err
 	}
 
-	dps, err := mc.getDeviceProfiles(bson.M{"commands": bson.M{"$elemMatch": query}})
+	dps, err := mc.getDeviceProfiles(bson.M{"commands.$id": command.Id})
```

Comment:
Before approving this PR, I want to document reasoning for the approval w/r/t the above line change.

The older version of the line above was introduced by @michaelestrin who was intending to use a Mongo projection to find all device profiles containing a command associated with the given ID. See [the docs](https://docs.mongodb.com/manual/reference/operator/query/elemMatch/) on that capability.

However that doesn't work if the incoming `command.Id` is a UUID because the array of commands on the device profile is an array of DBRefs, which means BSON IDs. Yet one more artifact of using Mongo as a faux relational database when we should just be storing documents.

Given that, it is necessary to obtain the command Mongo model so that we have the BSON ID and can then query for the relevant device profiles via the DBRef BSON ID.

Sample document for illustration:
```
{ 
"_id" : ObjectId("5c4f6e62deafc2723239b3a4"), 
"created" : NumberLong("1548709474557"), 
"modified" : NumberLong("1548709474557"), 
"origin" : NumberLong("1471806386919"), 
"description" : "BACnet Honeywell thermostats",
 "uuid" : "91b5e0c0-2752-43e0-9cd0-d9b23fd03eb2", 
"name" : "boiling point profile", 
"manufacturer" : "Honeywell", 
"model" : "ABC123", 
"labels" : [ "bacnet", "thermostat", "Honeywell" ], 
"objects" : 
    { "resources" : 
        { "CurrentHumidity" : 
            { "get" : [ 
                { "parameter" : "CurrentHumidity", "property" : "presentValue", "operation" : "get", "object" : "AnalogValue_22" }
                 ] 
            }
         }
    }, 
    "deviceResources" : [ ],
    "resources" : [ ], 
    "commands" : [
        DBRef("command", ObjectId("5c4f6e62deafc2723239b394")), 
        DBRef("command", ObjectId("5c4f6e62deafc2723239b39e")) 
    ] 
}
```